__tests__/codeEditor.test.tsx

```diff
@@ -1,12 +1,30 @@
 import { render, screen, waitFor } from '@testing-library/react';
 import React from 'react';
+import { afterEach, beforeEach, describe, expect, vi, it } from 'vitest';
 
 import CodeEditor from '../src/codeEditor';
 
 describe('<CodeEditor/>', () => {
-  const getClientRectSpy = jest.fn(() => ({ width: 100 }));
-  Range.prototype.getBoundingClientRect = getClientRectSpy;
-  Range.prototype.getClientRects = getClientRectSpy;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let originalGetBoundingClientRect: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let originalGetClientRects: any;
+  const getClientRectSpy = vi.fn(() => ({ width: 100 }));
```

Comment:
some eslint warnings here about the spy not being fully typed. probably safe to suppress?

Reply:
https://github.com/readmeio/syntax-highlighter/pull/595/commits/705b5dbd2e4c53d1cb3220a4798f73e51977eeea